tooling/hibernate-gradle-plugin/src/main/java/org/hibernate/orm/tooling/gradle/HibernateOrmPlugin.java

```diff
@@ -25,24 +25,28 @@
 public class HibernateOrmPlugin implements Plugin<Project> {
 	@Override
 	public void apply(Project project) {
-		// for SourceSet support and other JVM goodies
-		project.getPlugins().apply( JvmEcosystemPlugin.class );
+		project.getPlugins().withType( JavaPlugin.class, javaPlugin -> {
```

Comment:
I am no Gradle expert, and it's not an explicit goal to support other languages, I merely tried to reason about this change based on the naming of things, with the desire avoiding to loose functionality that was previously present. Given that all these other languages seem to rely on the Java Plugin, it seems like this change would be safe.

Reply:
Plugins applying other plugins is generally considered an anti-pattern.  Really, delayed configuration should be preferred when possible.  What is the exact issue you are seeing?

Reply:
If it's an anti-pattern, why is it so popular among the other plugins? 
Even the [gradle documentation ](https://docs.gradle.org/7.6.2/userguide/implementing_gradle_plugins.html#reacting_to_plugins)uses that in the samples. 

The issue I want to fix is that in the current state, the order of plugins in the build.gradle is important.
If the hibernate plugin is listed before the java plugin (or another plugin that applies the java plugin like Kotlin), the build stops with an error (as described in [HHH-15557](https://hibernate.atlassian.net/browse/HHH-15557)).

Reply:
I direct you to the Gradle Slack channel and you can ask that ;)

But the basic gist is that often you end up applying unintended/wrong features to the build

Reply:
I have read the docs again... the previous approach of using `project.getPlugins().apply( JvmEcosystemPlugin.class )` added the Plugin unconditionally.

The solution proposed here: `project.getPlugins().withType( JavaPlugin.class,...` should already do the right thing by only running the code in the Lambda if the JavaPlugin is present.

See the [JavaDoc](https://docs.gradle.org/current/javadoc/org/gradle/api/plugins/PluginManager.html#withPlugin(java.lang.String,org.gradle.api.Action)) for `project.getPlugins().withType`:

> Executes the given action when the specified plugin is applied.
If a plugin with the specified ID has already been applied, the supplied action will be executed immediately.  Otherwise, the action will executed immediately after a plugin with the specified ID is applied.  
The given action is always executed after the plugin has been applied.

So I would say that I am the right path.

Reply:
> Plugins applying other plugins is generally considered an anti-pattern.

No, it is not.
Plugins applying other plugins is quite normal.
If a plugin in any case requires another plugin to be applied, it is quite usual and normal to apply the other plugin.
But if a plugin just wants to do things in case another plugin is also applied at any time but also do sensible work if not, then reacting to the plugin application is better.

But `project.getPlugins()` is practically never the right thing to do, look at its JavaDoc.
To apply another plugin you would directly do `project.apply`,
to react to another plugin being applied you would use `project.getPluginManager().withPlugin("the-plugin-id", thePlugin -> ...)`.

> Even the gradle documentation uses that in the samples.

That's actually a very weak argument though. :-D
The docs show what is possible, not what is best-practice.
It shows quite some things you should avoid whereever possible.
But applying other plugins is not one of them.
Besides that this PR does exactly the opposite, it removes a plugin application and instead adds reacting to a plugin being applied.

> The issue I want to fix is that in the current state, the order of plugins in the build.gradle is important.

This indeed is bad practice. :-)

How to properly solve it I cannot tell as I don't know the context.
But I described above when you should use what.
So an alternative would then for example be to apply the `java` plugin instead of the `jvm-ecosystem` plugin, if the `java` plugin is a hard requirement for this plugin to work properly and would also solve the ordering problem.

Reply:
Thank you for the comments and suggestions, much appreciated!

I adjusted the commit to make use of `project.getPluginManager()`.

I think it is better to react on the Java plugin being added instead of adding it from the Gradle plugin.
All relevant language plugins apply the Java plugin by themselves, so there should be no need to do that in the Hibernate plugin. And if no JVM language has been loaded, there should be no classes that need transformation.